src/sagemaker/serve/builder/jumpstart_builder.py

```diff
@@ -437,42 +468,58 @@ def _build_for_jumpstart(self):
         self.secret_key = None
         self.jumpstart = True
 
-        pysdk_model = self._create_pre_trained_js_model()
+        self.pysdk_model = self._create_pre_trained_js_model()
+        self.pysdk_model.tune = lambda *args, **kwargs: self._default_tune()
 
-        image_uri = pysdk_model.image_uri
+        logger.info(
+            "JumpStart ID %s is packaged with Image URI: %s", self.model, self.pysdk_model.image_uri
+        )
 
-        logger.info("JumpStart ID %s is packaged with Image URI: %s", self.model, image_uri)
+        if self.mode != Mode.SAGEMAKER_ENDPOINT:
+            if self._is_gated_model(self.pysdk_model):
+                raise ValueError(
+                    "JumpStart Gated Models are only supported in SAGEMAKER_ENDPOINT mode."
+                )
 
-        if self._is_gated_model(pysdk_model) and self.mode != Mode.SAGEMAKER_ENDPOINT:
-            raise ValueError(
-                "JumpStart Gated Models are only supported in SAGEMAKER_ENDPOINT mode."
-            )
+            if "djl-inference" in self.pysdk_model.image_uri:
+                logger.info("Building for DJL JumpStart Model ID...")
+                self.model_server = ModelServer.DJL_SERVING
+                self.image_uri = self.pysdk_model.image_uri
 
-        if "djl-inference" in image_uri:
-            logger.info("Building for DJL JumpStart Model ID...")
-            self.model_server = ModelServer.DJL_SERVING
+                self._build_for_djl_jumpstart()
 
-            self.pysdk_model = pysdk_model
-            self.image_uri = self.pysdk_model.image_uri
+                self.pysdk_model.tune = self.tune_for_djl_jumpstart
+            elif "tgi-inference" in self.pysdk_model.image_uri:
+                logger.info("Building for TGI JumpStart Model ID...")
+                self.model_server = ModelServer.TGI
+                self.image_uri = self.pysdk_model.image_uri
 
-            self._build_for_djl_jumpstart()
+                self._build_for_tgi_jumpstart()
 
-            self.pysdk_model.tune = self.tune_for_djl_jumpstart
-        elif "tgi-inference" in image_uri:
-            logger.info("Building for TGI JumpStart Model ID...")
-            self.model_server = ModelServer.TGI
+                self.pysdk_model.tune = self.tune_for_tgi_jumpstart
+            elif "huggingface-pytorch-inference:" in self.pysdk_model.image_uri:
+                logger.info("Building for MMS JumpStart Model ID...")
+                self.model_server = ModelServer.TGI
```

Comment:
why mapping to TGI? should be MMS right?

Reply:
Fixed